ssz_serialization/codec.nim

```diff
@@ -185,54 +232,57 @@ func readSszValue*[T](input: openArray[byte],
     copyMem(addr val.bytes[0], unsafeAddr input[0], input.len)
 
   elif val is object|tuple:
-    let inputLen = uint32 input.len
-    const minimallyExpectedSize = uint32 fixedPortionSize(T)
-
-    if inputLen < minimallyExpectedSize:
-      raise newException(MalformedSszError, "SSZ input of insufficient size")
-
-    enumInstanceSerializedFields(val, fieldName, field):
-      const boundingOffsets = getFieldBoundingOffsets(T, fieldName)
-
-      # type FieldType = type field # buggy
-      # For some reason, Nim gets confused about the alias here. This could be a
-      # generics caching issue caused by the use of distinct types. Such an
-      # issue is very scary in general.
-      # The bug can be seen with the two List[uint64, N] types that exist in
-      # the spec, with different N.
-
-      type SszType = type toSszType(declval type(field))
-
-      when isFixedSize(SszType):
-        const
-          startOffset = boundingOffsets[0]
-          endOffset = boundingOffsets[1]
-      else:
-        let
-          startOffset = readOffsetUnchecked(boundingOffsets[0])
-          endOffset = if boundingOffsets[1] == -1: inputLen
-                      else: readOffsetUnchecked(boundingOffsets[1])
-
-        when boundingOffsets.isFirstOffset:
-          if startOffset != minimallyExpectedSize:
-            raise newException(MalformedSszError, "SSZ object dynamic portion starts at invalid offset")
-
-        if startOffset > endOffset:
-          raise newException(MalformedSszError, "SSZ field offsets are not monotonically increasing")
-        elif endOffset > inputLen:
-          raise newException(MalformedSszError, "SSZ field offset points past the end of the input")
-        elif startOffset < minimallyExpectedSize:
-          raise newException(MalformedSszError, "SSZ field offset points outside bounding offsets")
-
-      # TODO The extra type escaping here is a work-around for a Nim issue:
-      when type(field) is type(SszType):
-        readSszValue(
-          input.toOpenArray(int(startOffset), int(endOffset - 1)),
-          field)
-      else:
-        field = fromSszBytes(
-          type(field),
-          input.toOpenArray(int(startOffset), int(endOffset - 1)))
+    when type(val).isCaseObject():
```

Comment:
You should be able to use `isCaseObject(T)` instead of `type(val)` here.

Reply:
adjusted